viadot/sources/tm1.py

```diff
@@ -0,0 +1,182 @@
+            password=self.credentials["password"],
+            ssl=self.verify,
+        )
+
+    def get_cubes_names(self) -> list:
+        """
+        Get list of avaiable cubes in TM1 instance.
+
+        Returns:
+            list: List containing avaiable cubes names.
+
+        """
+        conn = self.get_connection()
+        return conn.cubes.get_all_names()
+
+    def get_views_names(self) -> list:
+        """
+        Get list of avaiable views in TM1 cube instance.
+
+        Returns:
+            list: List containing avaiable views names.
+
+        """
+        conn = self.get_connection()
+        return conn.views.get_all_names(self.cube)
+
+    def get_diemensions_names(self) -> list:
+        """
+        Get list of avaiable dimensions in TM1 instance.
+
+        Returns:
+            list: List containing avaiable dimensions names.
+
+        """
+        conn = self.get_connection()
+        return conn.dimensions.get_all_names()
+
+    def get_hierarchies_names(self) -> list:
+        """
+        Get list of avaiable hierarchies in TM1 dimension instance.
+
+        Returns:
+            list: List containing avaiable hierarchies names.
+
+        """
+        conn = self.get_connection()
+        return conn.hierarchies.get_all_names(self.dimension)
+
+    def get_available_elements(self) -> list:
+        """
+        Get list of avaiable elements in TM1 instance based on hierarchy and diemension.
+
+        Returns:
+            list: List containing avaiable elements names.
+
+        """
+        conn = self.get_connection()
+        return conn.elements.get_element_names(
+            dimension_name=self.dimension, hierarchy_name=self.hierarchy
+        )
+
+    def to_df(self, if_empty: Literal["warn", "fail", "skip"] = "skip") -> pd.DataFrame:
+        """
+        Function for downloading data from TM1 to pd.DataFrame.
+
+        Args:
+            if_empty (Literal["warn", "fail", "skip"], optional): What to do if output DataFrame is empty. Defaults to "skip".
+
+        Returns:
+            pd.DataFrame: DataFrame with data downloaded from TM1 view.
+
+        Raises:
+            ValidationError: When mdx and cube + view are not specified or when combination of both is specified.
+        """
+        conn = self.get_connection()
+
+        if self.mdx_query is None and (self.cube is None or self.view is None):
+            raise ValidationError("MDX query or cube and view are required.")
+        if self.cube is not None and self.view is not None:
+            df = conn.cubes.cells.execute_view_dataframe(
+                cube_name=self.cube,
+                view_name=self.view,
+                private=self.private,
+                top=self.limit,
+            )
+        elif self.mdx_query is not None:
+            df = conn.cubes.cells.execute_mdx_dataframe(self.mdx_query)
+        else:
+            raise ValidationError("Specify only one: MDX query or cube and view.")
```

Comment:
Please add information that is exactly needed to the class description. It will be helpful for the user who is not familiar with this class. 
The first explanation that you need `query` or `cube + view` is here in lines 164 and 175.